attachment_s3/controllers/main.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2021 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+import logging
+import os
+
+import odoo
+from odoo import SUPERUSER_ID
+from odoo.addons.web.controllers.main import Database
+from odoo import http
+from odoo import exceptions
+from odoo.http import request
+from ..s3uri import S3Uri
+
+_logger = logging.getLogger(__name__)
+
+
+class Database(Database):
+    @http.route()
+    def drop(self, master_pwd, name):
+        res = super().drop(master_pwd, name)
+        empty = os.environ.get("AWS_EMPTY_ON_DBDROP", False)
+        if empty:
+            try:
+                bucket = os.environ.get("AWS_BUCKETNAME").format(db=name)
+                bucket = request.env["ir.attachment"]._get_s3_bucket(bucket)
+                bucket.objects.all().delete()
+            except exceptions.UserError:
+                _logger.exception("Error deleting attachments in object storage.")
+        return res
+
+    @http.route()
+    def duplicate(self, master_pwd, name, new_name):
+        res = super().duplicate(master_pwd, name, new_name)
+        try:
```

Comment:
I would make this new duplication behaviour optional. May be with another environment variable?

Reply:
@jjscarafia ok